File: app/src/main/java/com/app/turingrobot/utils/DeviceUtils.java
package com.app.turingrobot.utils;

import android.content.Context;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;

/**
 * Created by Alpha on 2016/3/27 21:45.
 */
public class DeviceUtils {

    public static String getVersion(Context context) {
        PackageManager packageManager = context.getPackageManager();
        try {
            PackageInfo info = packageManager.getPackageInfo(context.getPackageName(), 0);
            return info.versionName;
        } catch (PackageManager.NameNotFoundException e) {
            e.printStackTrace();
            return "";
        }
    }
}

File: app/src/main/java/com/app/turingrobot/utils/DeviceUtils.kt
package com.app.turingrobot.utils

import android.content.Context
import android.content.pm.PackageInfo
import android.content.pm.PackageManager

/**
 * Created by Alpha on 2016/3/27 21:45.
 */
object DeviceUtils {

    fun getVersion(context: Context): String {
        val packageManager = context.packageManager
        try {
            val info = packageManager.getPackageInfo(context.packageName, 0)
            return info.versionName
        } catch (e: PackageManager.NameNotFoundException) {
            e.printStackTrace()
            return ""
        }

    }
}
